tpot/base.py

```diff
@@ -498,27 +537,60 @@ def _add_terminals(self, arg_types):
 
     def _setup_toolbox(self):
         with warnings.catch_warnings():
-            warnings.simplefilter('ignore')
-            creator.create('FitnessMulti', base.Fitness, weights=(-1.0, 1.0))
-            creator.create('Individual', gp.PrimitiveTree, fitness=creator.FitnessMulti, statistics=dict)
+            warnings.simplefilter("ignore")
+            creator.create("FitnessMulti", base.Fitness, weights=(-1.0, 1.0))
+            creator.create(
+                "Individual",
+                gp.PrimitiveTree,
+                fitness=creator.FitnessMulti,
+                statistics=dict,
+            )
 
         self._toolbox = base.Toolbox()
-        self._toolbox.register('expr', self._gen_grow_safe, pset=self._pset, min_=self._min, max_=self._max)
-        self._toolbox.register('individual', tools.initIterate, creator.Individual, self._toolbox.expr)
-        self._toolbox.register('population', tools.initRepeat, list, self._toolbox.individual)
-        self._toolbox.register('compile', self._compile_to_sklearn)
-        self._toolbox.register('select', tools.selNSGA2)
-        self._toolbox.register('mate', self._mate_operator)
+        self._toolbox.register(
+            "expr", self._gen_grow_safe, pset=self._pset, min_=self._min, max_=self._max
+        )
+        self._toolbox.register(
+            "individual", tools.initIterate, creator.Individual, self._toolbox.expr
+        )
+        self._toolbox.register(
+            "population", tools.initRepeat, list, self._toolbox.individual
+        )
+        self._toolbox.register("compile", self._compile_to_sklearn)
+        self._toolbox.register("select", tools.selNSGA2)
+        self._toolbox.register("mate", self._mate_operator)
         if self.tree_structure:
-            self._toolbox.register('expr_mut', self._gen_grow_safe, min_=self._min, max_=self._max + 1)
+            self._toolbox.register(
+                "expr_mut", self._gen_grow_safe, min_=self._min, max_=self._max + 1
+            )
         else:
-            self._toolbox.register('expr_mut', self._gen_grow_safe, min_=self._min, max_=self._max)
-        self._toolbox.register('mutate', self._random_mutation_operator)
+            self._toolbox.register(
+                "expr_mut", self._gen_grow_safe, min_=self._min, max_=self._max
+            )
+        self._toolbox.register("mutate", self._random_mutation_operator)
+
+    def _get_make_pipeline_func(self):
+        imblearn_used = (
+            Series(self.config_dict.keys())
```

Comment:
self.config_dict is None by default, please use `config_dict_` instead after calling [`self._setup_config(self.config_dict)`](https://github.com/EpistasisLab/tpot/blob/v0.11.6.post1/tpot/base.py#L526)